src/lib/route-builder.js

```diff
@@ -170,10 +174,13 @@ class RouteBuilder {
   /**
    * Round the hop's amounts according to the corresponding ledgers' scales/precisions.
    */
-  * _roundHop (hop, sourceUpDown, destinationUpDown) {
+  * _roundHop (hop, sourceUpDown, destinationUpDown, destinationPrecisionAndScale) {
     hop.sourceAmount = yield this._roundAmount('source', sourceUpDown, hop.sourceLedger, hop.sourceAmount)
-    hop.finalAmount = yield this._roundAmount('destination', destinationUpDown, hop.finalLedger, hop.finalAmount)
-    hop.destinationAmount = yield this._roundAmount('destination', destinationUpDown, hop.destinationLedger, hop.destinationAmount)
+    hop.destinationAmount = yield this._roundAmount('destination', destinationUpDown, hop.destinationLedger, hop.destinationAmount, destinationPrecisionAndScale)
+    // Only round the final amount if we know the precision/scale.
+    if (hop.isFinal || destinationPrecisionAndScale) {
```

Comment:
No, `hop.isFinal` implies that the destination ledger is adjacent the the connector, so the connector can use the appropriate ledger plugin the discover the precision & scale.

`destinationPrecisionAndScale` only needs to be provided when the connector _cannot_ discover the precision & scale.